File: homework/HW1/Palindrome.java
package homework.HW1;


public class Palindrome{

    public static boolean isPalindrome(String s){

        s = s.toLowerCase();
        s = s.replaceAll("[^a-z0-9]", "");
        //System.out.println(s);
        int length = s.length();
        boolean palindrome = true;

        for(int i = 0; i < length/2; i++){
            if(s.charAt(i) != s.charAt(length - 1 - i)){
                palindrome = false; 
                break;
            }
        }
        return palindrome;
    }

    public static void main(String[] args) {
        String myPalindrome = "#^!/race car";

        System.out.println("Palindrome: " + isPalindrome(myPalindrome));
    }


}

File: homework/HW1/Palindrome.py
import re
class Palindrome:

    def isPalindrome(s):
        s = s.lower()
        s = re.sub("[^a-z0-9]", "", s)
        print(s)
        length = len(s)
        palindrome = True

        for i in range(length//2):
            if s[i] != s[length -1 - i]:
                palindrome = False
                break
        
        return palindrome
    
myPalindrome = "@#!aaaBbaaa"
print("Palindrome: ", Palindrome.isPalindrome(myPalindrome))
        
